src/sqlancer/common/ast/newast/Select.java

```diff
@@ -0,0 +1,48 @@
+package sqlancer.common.ast.newast;
+
+import java.util.List;
+
+import sqlancer.common.schema.AbstractTable;
+import sqlancer.common.schema.AbstractTableColumn;
+
+public interface Select<J extends Join<E, T, C>, E extends Expression<C>, T extends AbstractTable<C, ?, ?>, C extends AbstractTableColumn<?, ?>>
+        extends Expression<C> {
+
+    List<E> getFromList();
+
+    void setFromList(List<E> fromList);
+
+    Expression<C> getWhereClause();
+
+    void setWhereClause(E whereClause);
+
+    void setGroupByClause(List<E> groupByClause);
+
+    List<E> getGroupByClause();
+
+    void setLimitClause(E limitClause);
+
+    Expression<C> getLimitClause();
+
+    List<E> getOrderByClause();
+
+    void setOrderByExpressions(List<E> orderBy);
```

Comment:
Here, should we try to use uniform naming? We could use `setOrderByClause` here or `getOrderByExpressions` above.